src/Init/Data/Int/Bitwise/Lemmas.lean

```diff
@@ -0,0 +1,59 @@
+/-
+Copyright (c) 2023 Siddharth Bhat. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Siddharth Bhat, Jeremy Avigad
+-/
+prelude
+import Init.Data.Nat.Bitwise.Lemmas
+import Init.Data.Int.Bitwise
+
+namespace Int
+
+theorem shiftRight_eq (n : Int) (s : Nat) : n >>> s = Int.shiftRight n s := rfl
+@[simp]
+theorem shiftRight_ofNat (n s : Nat) : (n : Int) >>> s = Int.ofNat (n >>> s) := rfl
+theorem natCast_shiftRight (n s : Nat) : ((↑n) : Int) >>> s = n >>> s := rfl
+
+@[simp]
+theorem shiftRight_negSucc (m n : Nat) :
+    -[m+1] >>> n = -[m >>>n +1] := rfl
```

Comment:
`>>>` is an infix operator, so this should be named `negSucc_shiftRight`.

But my general object remains, I'd really prefer to mention `negSucc` as little as possible. :-(

Reply:
@semorrison I must admit, I don't understand the reason for shying away from writing this `simp` lemma here: It's building the basic theory of `>>>`, and this is taken [straight from mathlib](https://github.com/leanprover-community/mathlib4/blob/c9ee63a2a9d4eeb41607c930473a224b081dc2c9/Mathlib/Data/Int/Bitwise.lean#L490-L490)

Reply:
I've fixed the name.

Reply:
I just don't want people using `Int.negSucc` when they don't need to, because it's a slippery slope of ever increasing API surface area. It's meant to be an internal implementation detail of `Int`. What if we decided to change the definition of `Int` to pairs of `Nat`s modulo `(a,b) ~ (a+c, b+c)`? (Obviously we won't...)